utils/oscap-run-sce-script

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+#
+# Authors:
+#      Dominique Blaze <contact@d0m.tech>
+#
+# use by oscap for evaluate a SCE file when +x rights are missing
+
+if [ ! -z $1 ] && [ -f $1 ] 
+then
+	# file exists. first check if shebang is here 
+	
+	firstline=$(head -n1 $1)
+	if [ ${firstline:0:2} = "#!" ]
+	then  # it's a shebang
+		cmd=${firstline:2}  # remove the begin (#!)
+		cmd=${cmd##*( )}  # trim whitespaces
+		eval $cmd $1 > /dev/stdout
```

Comment:
How do we know that we don't run arbitrary binary?

Reply:
Isn't it the point to run some arbitrary binary? Or you expect that this wrapper could be abused?

Reply:
Hello,
In the case were oscap-run-sce-script is used directly by the user, the privileges are exactly the same, there is no special permissions accorded here. And if the cmd contains an arbitrary "binary", it will not be executed if it have not +x right (if if it have it, the user can run it directly).

Example, chmod -x on /usr/bin/lsub, create a .sh file in tmp with the shebang #! /usr/bin/lsusb, run oscap-run-sce-script /tmp/tmp.sh, it will try to eval /usr/bin/lsusb but this binary will not be executed.

The only case where an "privilege escalation" could be observed is if the user have no bash access, but is able to run arbitrarily oscap-run-sce-script, and not that if he can do that he can probably also run oscap on a custom datastream containing bash commands. So yes, I agree that there is a possibility of "bash injection", but in our case i don't see the risk since there is no privilege escalation (when using directly oscap-run-sce-script), and moreover this "bash injection" can also be done by evaluating a custom datastream, so for me there is no more risks with or without oscap-run-sce-script.

The only potential security issue that I see is if oscap is running with root privileges, we need to be absolutely sure that the file evaluated (previously extracted from the DS) is safe, i.e. the DS should be signed and we should ensure that the content of the tmp file haven't been modified between its extraction and it's launch; I've not analyse more this potential issue, but the issue is exactly the same with or without oscap-run-sce-script (because oscap-run-sce-script can have root permissions only if launched from oscap running as root). But this subject should be in the scope on another PR (and i think it should be a good idea to discuss about "should we run sce script as root if oscap is running as root ? should we introduce a specific attribute specifying if the sce script should be ran as root ? Perhaps this possibility should be allowed only with DSIG enabled on the DS").

I don't see any other way of "abusing" the oscap-run-sce-script, but feel free to correct me if i'm going wrong and to suggest a mitigation (note that originally this wrapper script if an idea of martin preisler, it's not mandatory to use it but easy to use and/or customize, and currently I don't see in what it adds a security issue).

Reply:
To bring a bit more context to the topic: https://serverfault.com/questions/72356/how-useful-is-mounting-tmp-noexec

So, in general I agree with @DominiqueDevinci about it not increasing the surface of attack in any way, because there is already a possibility to run scripts from data streams.

On the other hand, if you would prefer to not to execute arbitrary binaries as interpreters, the only way to go it is to create a "white list" right in the wrapper. Does it look like feasible and/or sensible, @jan-cerny?

Reply:
Do you think that attackers could write their payload to `/tmp` and then call our wrapper to execute the payload if they find that there is `noexec` set on `/tmp`?

Reply:
@evgenyz about the context I agree that the gain of /tmp with noexec mode isn't huge.
However, it's a specification (in my case it's required by ANSSI NT28.R12), and even if we can  disagree with that, the goal here is to be compliant, and it's not possible currently with openscap using SCE (and SCE is incredibly usefull for custom advanced checks). I think that's the most important point of the context.

A white list seems a correct mitigation for me (the real question is to agree with SCE "specifications", if exists ... users should be informed of this white list), and so we should use only this wrapper (even if the tmp file is executable). I will investigate later about the risks of running oscap as root with SCE, because it's an importante question.

Reply:
It won't work as /tmp is noexec, and it won't allow to have anything executable there. And eval won't execute it.